File: Samples/Intent.Sample.Angular/Intent.Sample.Angular/src/app/service-proxies/models/backend/services/customers/customer-dto.ts
import { CustomerTitleDto } from './customer-title-dto';
import { CustomerAddressDto } from './customer-address-dto';
import { CustomerLoyaltyDto } from './customer-loyalty-dto';


export interface CustomerDto {
  id: string;
  titleId: string;
  name: string;
  surname: string;
  email: string;
  isActive: boolean;
  title: CustomerTitleDto;
  addresses: CustomerAddressDto[];
  loyalty: CustomerLoyaltyDto | null;
}

File: Samples/Intent.Sample.Blazor.Server/Intent.Sample.Blazor.Server/Intent.Sample.Blazor.Server.Application/Customers/CustomerDto.cs
using Intent.RoslynWeaver.Attributes;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Dtos.DtoModel", Version = "1.0")]

namespace Intent.Sample.Blazor.Server.Application.Customers
{
    public record CustomerDto
    {
        public CustomerDto()
        {
            Name = null!;
            Surname = null!;
            Email = null!;
            Title = null!;
            Addresses = null!;
        }

        public Guid Id { get; init; }
        public Guid TitleId { get; init; }
        public string Name { get; init; }
        public string Surname { get; init; }
        public string Email { get; init; }
        public bool IsActive { get; init; }
        public CustomerTitleDto Title { get; init; }
        public List<CustomerAddressDto> Addresses { get; init; }
        public CustomerLoyaltyDto? Loyalty { get; init; }
    }
}